### PR TITLE
Remove the old REPL tab completion implementation

### PR DESCRIPTION
**Pull request by retronym**
We kept it around behind an option in 2.11.x, but we can
jettison it in 2.12.x.

Review by @lrytz

### CODE REVIEW

File: src/repl-jline/scala/tools/nsc/interpreter/jline/JLineReader.scala [hunk 1]
```diff
@@ -125,21 +125,16 @@ private class JLineConsoleReader extends jconsole.ConsoleReader with interpreter
     // adapt the JLine completion interface
     def completer =
       new Completer {
-        val tc = completion.completer()
+        val tc = completion
```
Comment (lrytz) [inline src/repl-jline/scala/tools/nsc/interpreter/jline/JLineReader.scala:128]: tc is unused now

**Comment by lrytz** [comment]
LGTM

**Comment by som-snytt** [comment]
Shouldn't there be some sort of ceremony? And something with wine and cheese afterwards?

**Comment by retronym** [comment]
:wine_glass: :fork_and_knife:

**Pull request merged**